spectec/spec/7-module.watsup

```diff
@@ -0,0 +1,236 @@
+;; Allocation
+
+def $alloc_import(moduleinst, import*, externval*) : moduleinst
+
+def $alloc_import(m, epsilon, epsilon) = m
+
+def $alloc_import(m, import import'*, externval externval'*) = m_res
```

Comment:
Perhaps refactor this into one case that generically deals with sequences of imports (longer than 1), and separate cases for singular imports of specific form. That would also make it easier to drop some of the side conditions in favour of pattern matching.